python/packages/azurefunctions/agent_framework_azurefunctions/_entities.py

```diff
@@ -89,39 +137,54 @@ async def run_agent(
 
         message = run_request.message
         thread_id = run_request.thread_id
-        correlation_id = run_request.correlation_id
+        correlationId = run_request.correlationId
         if not thread_id:
             raise ValueError("RunRequest must include a thread_id")
-        if not correlation_id:
-            raise ValueError("RunRequest must include a correlation_id")
+        if not correlationId:
+            raise ValueError("RunRequest must include a correlationId")
         role = run_request.role or Role.USER
         response_format = run_request.response_format
         enable_tool_calls = run_request.enable_tool_calls
 
+        state_request = DurableAgentStateRequest.from_run_request(run_request)
+        self.state.data.conversationHistory.append(state_request)
+
         logger.debug(f"[AgentEntity.run_agent] Received message: {message}")
         logger.debug(f"[AgentEntity.run_agent] Thread ID: {thread_id}")
-        logger.debug(f"[AgentEntity.run_agent] Correlation ID: {correlation_id}")
+        logger.debug(f"[AgentEntity.run_agent] Correlation ID: {correlationId}")
         logger.debug(f"[AgentEntity.run_agent] Role: {role.value}")
         logger.debug(f"[AgentEntity.run_agent] Enable tool calls: {enable_tool_calls}")
         logger.debug(f"[AgentEntity.run_agent] Response format: {'provided' if response_format else 'none'}")
-
-        # Store message in history with role
-        self.state.add_user_message(message, role=role, correlation_id=correlation_id)
-
+        logger.debug(f"[AgentEntity.run_agent] Saved state request: {state_request}")
         logger.debug("[AgentEntity.run_agent] Executing agent...")
 
         try:
             logger.debug("[AgentEntity.run_agent] Starting agent invocation")
 
-            run_kwargs: dict[str, Any] = {"messages": self.state.get_chat_messages()}
+            # Build messages from conversation history, excluding error responses
+            # Error responses are kept in history for tracking but not sent to the agent
+            chat_messages: list[Any] = [
+                m.to_chat_message()
+                for entry in self.state.data.conversationHistory
+                if not self._is_error_response(entry)
+                for m in entry.messages
+            ]
+
+            # Strip additional_properties from all messages to avoid metadata being sent to Azure OpenAI
```

Comment:
This is a bug in agent-framework only for ResponsesAPI (the same thing works for ChatCompletion API), we shouldn't strip our data until we get confirmation from them on best practices on how to proceed here #2203

Reply:
For now since we can't proceed without stripping additional properties, I'm keeping it the same. I'll add a follow up task to fix this once agent-framework comes with an alternative

Reply:
ChatCompletion API works though and since we're using those primarily for our samples we should not do any additional data mucking on our end. We can always add this in if they say its not supported. I would rather not do this just for the ResponsesAPI.

Reply:
Removed in #2315